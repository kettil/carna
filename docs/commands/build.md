# The `build` task

Build the application

## Options

| Options | Description       |
| ------- | ----------------- |
| --ci    | Run it in CI mode |

## Exit codes

| Exit code | Description |
| --------- | ----------- |
| 0         | Success     |
| 1         | task failed |
