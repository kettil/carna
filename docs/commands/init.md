# The `init` task

Creates a new project with all config files in the current folder.

## Options

| Options    | short | Description                                                  |
| ---------- | ----- | ------------------------------------------------------------ |
| --package  | -p    | Project is created as a package                              |
| --cli      | -c    | Extends the package with CLI features (required `--package`) |
| --noCommit |       | No initial commit is executed at the end                     |

## Exit codes

| Exit code | Description |
| --------- | ----------- |
| 0         | Success     |
| 1         | task failed |
