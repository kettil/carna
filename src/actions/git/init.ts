import { join } from 'path';
import access from '../../lib/cmd/access';
import exec from '../../lib/cmd/exec';
import { Action } from '../../lib/types';

const gitInit: Action = async ({ cwd, log }) => {
  const path = join(cwd, '.git');
  const isExists = await access(path);

  if (!isExists) {
    log.info('Initialize the Git repository');
    await exec({ cmd: 'git', args: ['init'], cwd, log });
  } else {
    log.info('Git repository already exists');
  }
};

export default gitInit;
