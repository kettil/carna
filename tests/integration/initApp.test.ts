import * as fs from 'fs';
import { join } from 'path';
import { handler } from '../../src/lib/commands/init';
import {
  cwd,
  getAccessFiles,
  getArgv,
  getReaddirFiles,
  getReadFileFiles,
  getReadFileWithHooksFiles,
} from '../shared/configs';

jest.mock('child_process', () => require('../shared/__mock__/childProcess'));
jest.mock('fs', () => require('../shared/__mock__/fs'));
jest.mock('ora', () => require('../shared/__mock__/ora'));
jest.mock('exit', () => require('../shared/__mock__/exit'));
jest.mock('semver', () => require('../shared/__mock__/semver'));
jest.mock('depcheck', () => require('../shared/__mock__/depcheck'));
jest.mock('istanbul-reports', () => require('../shared/__mock__/istanbulReports'));

describe('command init (app)', () => {
  beforeEach(() => {
    (fs as any).setMockAccessFiles(
      getAccessFiles({
        [join(cwd, '.vscode')]: false,
        [join(cwd, 'src')]: false,
        [join(cwd, 'src/bin')]: false,
        [join(cwd, 'src/lib')]: false,
        [join(cwd, 'src/lib/types.ts')]: false,
        [join(cwd, 'tests')]: false,
        [join(cwd, 'tests', 'shared')]: false,
        [join(cwd, 'tests', 'shared', '.keep')]: false,
        [join(cwd, 'tests', 'type')]: false,
        [join(cwd, 'tests', 'type', '.keep')]: false,
        [join(cwd, 'tests', 'unit')]: false,
        [join(cwd, 'tests', 'integration')]: false,
        [join(cwd, 'tests', 'e2e')]: false,
      }),
    );

    (fs as any).setMockReadFileFiles(getReadFileFiles());
    (fs as any).setMockReaddirFiles(getReaddirFiles());
  });

  test('it should work with init commit', async () => {
    const result = await handler(getArgv({ package: false, cli: false, noCommit: false }));

    expect(result).toBeUndefined();
  });

  test('it should work without init commit', async () => {
    const result = await handler(getArgv({ package: false, cli: false, noCommit: true }));

    expect(result).toBeUndefined();
  });

  test('it should work with hooks and with init commit', async () => {
    (fs as any).setMockReadFileFiles(getReadFileWithHooksFiles());

    const result = await handler(getArgv({ package: false, cli: false, noCommit: false }));

    expect(result).toBeUndefined();
  });

  test('it should work with hooks and without init commit', async () => {
    (fs as any).setMockReadFileFiles(getReadFileWithHooksFiles());

    const result = await handler(getArgv({ package: false, cli: false, noCommit: true }));

    expect(result).toBeUndefined();
  });
});
