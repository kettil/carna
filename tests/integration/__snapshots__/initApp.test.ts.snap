// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command init (app) it should work with hooks and with init commit: child-process 1`] = `
Object {
  "args": Array [
    "init",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with hooks and with init commit: child-process 2`] = `
Object {
  "args": Array [
    "install",
    "--save-dev",
    "--save-exact",
    "@types/node",
    "carna",
    "typescript",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with hooks and with init commit: child-process 3`] = `
Object {
  "args": Array [
    "run",
    "prepare",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with hooks and with init commit: child-process 4`] = `
Object {
  "args": Array [
    "add",
    ".",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with hooks and with init commit: child-process 5`] = `
Object {
  "args": Array [
    "commit",
    "-m",
    "\\"feat: üê£\\"",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 1`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/.vscode",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 2`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/src",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 3`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/src/lib",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 4`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/shared",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 5`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/type",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 6`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/unit",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 7`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/integration",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 8`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/e2e",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 9`] = `
Object {
  "data": "
",
  "path": "/path/to/project/src/lib/types.ts",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 10`] = `
Object {
  "data": "
",
  "path": "/path/to/project/tests/shared/.keep",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 11`] = `
Object {
  "data": "
",
  "path": "/path/to/project/tests/type/.keep",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 12`] = `
Object {
  "data": "{\\"file\\":\\"settings.json\\"}
",
  "path": "/path/to/project/.vscode/settings.json",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 13`] = `
Object {
  "data": "{\\"file\\":\\"carnarc.json\\"}
",
  "path": "/path/to/project/.carnarc.json",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 14`] = `
Object {
  "data": "editorconfig
",
  "path": "/path/to/project/.editorconfig",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 15`] = `
Object {
  "data": "eslintignore
",
  "path": "/path/to/project/.eslintignore",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 16`] = `
Object {
  "data": "gitignore
",
  "path": "/path/to/project/.gitignore",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 17`] = `
Object {
  "data": "npmignore
",
  "path": "/path/to/project/.npmignore",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 18`] = `
Object {
  "data": "prettierignore
",
  "path": "/path/to/project/.prettierignore",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 19`] = `
Object {
  "data": "// babel.config.js
",
  "path": "/path/to/project/babel.config.js",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 20`] = `
Object {
  "data": "{\\"file\\":\\"typescriptrc.json\\"}
",
  "path": "/path/to/project/tsconfig.json",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 21`] = `
Object {
  "data": "{\\"file\\":\\"typescriptrc.build.json\\"}
",
  "path": "/path/to/project/tsconfig.build.json",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 22`] = `
Object {
  "data": "// jest.config.js
",
  "path": "/path/to/project/jest.config.js",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 23`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/unit/dummy.test.ts",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 24`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/integration/dummy.test.ts",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 25`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/e2e/dummy.test.ts",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 26`] = `
Object {
  "data": "// dummyHook.test.ts
",
  "path": "/path/to/project/tests/e2e/pre.ts",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 27`] = `
Object {
  "data": "// dummyHook.test.ts
",
  "path": "/path/to/project/tests/e2e/post.ts",
}
`;

exports[`command init (app) it should work with hooks and with init commit: fs 28`] = `
Object {
  "data": "{
  \\"name\\": \\"project-name\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"precarna:analyse\\": \\"echo \\\\\\"pre analyse hook is called\\\\\\"\\",
    \\"postcarna:analyse\\": \\"echo \\\\\\"post analyse hook is called\\\\\\"\\",
    \\"precarna:build\\": \\"echo \\\\\\"pre build hook is called\\\\\\"\\",
    \\"postcarna:build\\": \\"echo \\\\\\"post build hook is called\\\\\\"\\",
    \\"precarna:ci\\": \\"echo \\\\\\"pre ci hook is called\\\\\\"\\",
    \\"postcarna:ci\\": \\"echo \\\\\\"post ci hook is called\\\\\\"\\",
    \\"precarna:deps\\": \\"echo \\\\\\"pre deps hook is called\\\\\\"\\",
    \\"postcarna:deps\\": \\"echo \\\\\\"post deps hook is called\\\\\\"\\",
    \\"precarna:git\\": \\"echo \\\\\\"pre git hook is called\\\\\\"\\",
    \\"postcarna:git\\": \\"echo \\\\\\"post git hook is called\\\\\\"\\",
    \\"precarna:license\\": \\"echo \\\\\\"pre license hook is called\\\\\\"\\",
    \\"postcarna:license\\": \\"echo \\\\\\"post license hook is called\\\\\\"\\",
    \\"precarna:start\\": \\"echo \\\\\\"pre start hook is called\\\\\\"\\",
    \\"postcarna:start\\": \\"echo \\\\\\"post start hook is called\\\\\\"\\",
    \\"precarna:test\\": \\"echo \\\\\\"pre test hook is called\\\\\\"\\",
    \\"postcarna:test\\": \\"echo \\\\\\"post test hook is called\\\\\\"\\",
    \\"prepare\\": \\"[ \\\\\\"$CI\\\\\\" != \\\\\\"\\\\\\" ] || husky install ./node_modules/carna/configs/husky\\",
    \\"analyse\\": \\"npx carna analyse\\",
    \\"build\\": \\"npx carna build\\",
    \\"prebuild\\": \\"rm -rf ./build\\",
    \\"test\\": \\"npx carna test\\",
    \\"test:watch\\": \\"npx carna test -w\\"
  },
  \\"dependencies\\": {
    \\"package\\": \\"0.1.0\\"
  },
  \\"main\\": \\"build/index.js\\",
  \\"types\\": \\"build/index.d.ts\\"
}
",
  "path": "/path/to/project/package.json",
}
`;

exports[`command init (app) it should work with hooks and without init commit: child-process 1`] = `
Object {
  "args": Array [
    "init",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with hooks and without init commit: child-process 2`] = `
Object {
  "args": Array [
    "install",
    "--save-dev",
    "--save-exact",
    "@types/node",
    "carna",
    "typescript",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with hooks and without init commit: child-process 3`] = `
Object {
  "args": Array [
    "run",
    "prepare",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 1`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/.vscode",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 2`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/src",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 3`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/src/lib",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 4`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/shared",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 5`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/type",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 6`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/unit",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 7`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/integration",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 8`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/e2e",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 9`] = `
Object {
  "data": "
",
  "path": "/path/to/project/src/lib/types.ts",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 10`] = `
Object {
  "data": "
",
  "path": "/path/to/project/tests/shared/.keep",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 11`] = `
Object {
  "data": "
",
  "path": "/path/to/project/tests/type/.keep",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 12`] = `
Object {
  "data": "{\\"file\\":\\"settings.json\\"}
",
  "path": "/path/to/project/.vscode/settings.json",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 13`] = `
Object {
  "data": "{\\"file\\":\\"carnarc.json\\"}
",
  "path": "/path/to/project/.carnarc.json",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 14`] = `
Object {
  "data": "editorconfig
",
  "path": "/path/to/project/.editorconfig",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 15`] = `
Object {
  "data": "eslintignore
",
  "path": "/path/to/project/.eslintignore",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 16`] = `
Object {
  "data": "gitignore
",
  "path": "/path/to/project/.gitignore",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 17`] = `
Object {
  "data": "npmignore
",
  "path": "/path/to/project/.npmignore",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 18`] = `
Object {
  "data": "prettierignore
",
  "path": "/path/to/project/.prettierignore",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 19`] = `
Object {
  "data": "// babel.config.js
",
  "path": "/path/to/project/babel.config.js",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 20`] = `
Object {
  "data": "{\\"file\\":\\"typescriptrc.json\\"}
",
  "path": "/path/to/project/tsconfig.json",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 21`] = `
Object {
  "data": "{\\"file\\":\\"typescriptrc.build.json\\"}
",
  "path": "/path/to/project/tsconfig.build.json",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 22`] = `
Object {
  "data": "// jest.config.js
",
  "path": "/path/to/project/jest.config.js",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 23`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/unit/dummy.test.ts",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 24`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/integration/dummy.test.ts",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 25`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/e2e/dummy.test.ts",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 26`] = `
Object {
  "data": "// dummyHook.test.ts
",
  "path": "/path/to/project/tests/e2e/pre.ts",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 27`] = `
Object {
  "data": "// dummyHook.test.ts
",
  "path": "/path/to/project/tests/e2e/post.ts",
}
`;

exports[`command init (app) it should work with hooks and without init commit: fs 28`] = `
Object {
  "data": "{
  \\"name\\": \\"project-name\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"precarna:analyse\\": \\"echo \\\\\\"pre analyse hook is called\\\\\\"\\",
    \\"postcarna:analyse\\": \\"echo \\\\\\"post analyse hook is called\\\\\\"\\",
    \\"precarna:build\\": \\"echo \\\\\\"pre build hook is called\\\\\\"\\",
    \\"postcarna:build\\": \\"echo \\\\\\"post build hook is called\\\\\\"\\",
    \\"precarna:ci\\": \\"echo \\\\\\"pre ci hook is called\\\\\\"\\",
    \\"postcarna:ci\\": \\"echo \\\\\\"post ci hook is called\\\\\\"\\",
    \\"precarna:deps\\": \\"echo \\\\\\"pre deps hook is called\\\\\\"\\",
    \\"postcarna:deps\\": \\"echo \\\\\\"post deps hook is called\\\\\\"\\",
    \\"precarna:git\\": \\"echo \\\\\\"pre git hook is called\\\\\\"\\",
    \\"postcarna:git\\": \\"echo \\\\\\"post git hook is called\\\\\\"\\",
    \\"precarna:license\\": \\"echo \\\\\\"pre license hook is called\\\\\\"\\",
    \\"postcarna:license\\": \\"echo \\\\\\"post license hook is called\\\\\\"\\",
    \\"precarna:start\\": \\"echo \\\\\\"pre start hook is called\\\\\\"\\",
    \\"postcarna:start\\": \\"echo \\\\\\"post start hook is called\\\\\\"\\",
    \\"precarna:test\\": \\"echo \\\\\\"pre test hook is called\\\\\\"\\",
    \\"postcarna:test\\": \\"echo \\\\\\"post test hook is called\\\\\\"\\",
    \\"prepare\\": \\"[ \\\\\\"$CI\\\\\\" != \\\\\\"\\\\\\" ] || husky install ./node_modules/carna/configs/husky\\",
    \\"analyse\\": \\"npx carna analyse\\",
    \\"build\\": \\"npx carna build\\",
    \\"prebuild\\": \\"rm -rf ./build\\",
    \\"test\\": \\"npx carna test\\",
    \\"test:watch\\": \\"npx carna test -w\\"
  },
  \\"dependencies\\": {
    \\"package\\": \\"0.1.0\\"
  },
  \\"main\\": \\"build/index.js\\",
  \\"types\\": \\"build/index.d.ts\\"
}
",
  "path": "/path/to/project/package.json",
}
`;

exports[`command init (app) it should work with init commit: child-process 1`] = `
Object {
  "args": Array [
    "init",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with init commit: child-process 2`] = `
Object {
  "args": Array [
    "install",
    "--save-dev",
    "--save-exact",
    "@types/node",
    "carna",
    "typescript",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with init commit: child-process 3`] = `
Object {
  "args": Array [
    "run",
    "prepare",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with init commit: child-process 4`] = `
Object {
  "args": Array [
    "add",
    ".",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with init commit: child-process 5`] = `
Object {
  "args": Array [
    "commit",
    "-m",
    "\\"feat: üê£\\"",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work with init commit: fs 1`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/.vscode",
}
`;

exports[`command init (app) it should work with init commit: fs 2`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/src",
}
`;

exports[`command init (app) it should work with init commit: fs 3`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/src/lib",
}
`;

exports[`command init (app) it should work with init commit: fs 4`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/shared",
}
`;

exports[`command init (app) it should work with init commit: fs 5`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/type",
}
`;

exports[`command init (app) it should work with init commit: fs 6`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/unit",
}
`;

exports[`command init (app) it should work with init commit: fs 7`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/integration",
}
`;

exports[`command init (app) it should work with init commit: fs 8`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/e2e",
}
`;

exports[`command init (app) it should work with init commit: fs 9`] = `
Object {
  "data": "
",
  "path": "/path/to/project/src/lib/types.ts",
}
`;

exports[`command init (app) it should work with init commit: fs 10`] = `
Object {
  "data": "
",
  "path": "/path/to/project/tests/shared/.keep",
}
`;

exports[`command init (app) it should work with init commit: fs 11`] = `
Object {
  "data": "
",
  "path": "/path/to/project/tests/type/.keep",
}
`;

exports[`command init (app) it should work with init commit: fs 12`] = `
Object {
  "data": "{\\"file\\":\\"settings.json\\"}
",
  "path": "/path/to/project/.vscode/settings.json",
}
`;

exports[`command init (app) it should work with init commit: fs 13`] = `
Object {
  "data": "{\\"file\\":\\"carnarc.json\\"}
",
  "path": "/path/to/project/.carnarc.json",
}
`;

exports[`command init (app) it should work with init commit: fs 14`] = `
Object {
  "data": "editorconfig
",
  "path": "/path/to/project/.editorconfig",
}
`;

exports[`command init (app) it should work with init commit: fs 15`] = `
Object {
  "data": "eslintignore
",
  "path": "/path/to/project/.eslintignore",
}
`;

exports[`command init (app) it should work with init commit: fs 16`] = `
Object {
  "data": "gitignore
",
  "path": "/path/to/project/.gitignore",
}
`;

exports[`command init (app) it should work with init commit: fs 17`] = `
Object {
  "data": "npmignore
",
  "path": "/path/to/project/.npmignore",
}
`;

exports[`command init (app) it should work with init commit: fs 18`] = `
Object {
  "data": "prettierignore
",
  "path": "/path/to/project/.prettierignore",
}
`;

exports[`command init (app) it should work with init commit: fs 19`] = `
Object {
  "data": "// babel.config.js
",
  "path": "/path/to/project/babel.config.js",
}
`;

exports[`command init (app) it should work with init commit: fs 20`] = `
Object {
  "data": "{\\"file\\":\\"typescriptrc.json\\"}
",
  "path": "/path/to/project/tsconfig.json",
}
`;

exports[`command init (app) it should work with init commit: fs 21`] = `
Object {
  "data": "{\\"file\\":\\"typescriptrc.build.json\\"}
",
  "path": "/path/to/project/tsconfig.build.json",
}
`;

exports[`command init (app) it should work with init commit: fs 22`] = `
Object {
  "data": "// jest.config.js
",
  "path": "/path/to/project/jest.config.js",
}
`;

exports[`command init (app) it should work with init commit: fs 23`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/unit/dummy.test.ts",
}
`;

exports[`command init (app) it should work with init commit: fs 24`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/integration/dummy.test.ts",
}
`;

exports[`command init (app) it should work with init commit: fs 25`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/e2e/dummy.test.ts",
}
`;

exports[`command init (app) it should work with init commit: fs 26`] = `
Object {
  "data": "// dummyHook.test.ts
",
  "path": "/path/to/project/tests/e2e/pre.ts",
}
`;

exports[`command init (app) it should work with init commit: fs 27`] = `
Object {
  "data": "// dummyHook.test.ts
",
  "path": "/path/to/project/tests/e2e/post.ts",
}
`;

exports[`command init (app) it should work with init commit: fs 28`] = `
Object {
  "data": "{
  \\"name\\": \\"project-name\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"prepare\\": \\"[ \\\\\\"$CI\\\\\\" != \\\\\\"\\\\\\" ] || husky install ./node_modules/carna/configs/husky\\",
    \\"analyse\\": \\"npx carna analyse\\",
    \\"build\\": \\"npx carna build\\",
    \\"prebuild\\": \\"rm -rf ./build\\",
    \\"test\\": \\"npx carna test\\",
    \\"test:watch\\": \\"npx carna test -w\\"
  },
  \\"dependencies\\": {
    \\"package\\": \\"0.1.0\\"
  },
  \\"main\\": \\"build/index.js\\",
  \\"types\\": \\"build/index.d.ts\\"
}
",
  "path": "/path/to/project/package.json",
}
`;

exports[`command init (app) it should work without init commit: child-process 1`] = `
Object {
  "args": Array [
    "init",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work without init commit: child-process 2`] = `
Object {
  "args": Array [
    "install",
    "--save-dev",
    "--save-exact",
    "@types/node",
    "carna",
    "typescript",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work without init commit: child-process 3`] = `
Object {
  "args": Array [
    "run",
    "prepare",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command init (app) it should work without init commit: fs 1`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/.vscode",
}
`;

exports[`command init (app) it should work without init commit: fs 2`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/src",
}
`;

exports[`command init (app) it should work without init commit: fs 3`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/src/lib",
}
`;

exports[`command init (app) it should work without init commit: fs 4`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/shared",
}
`;

exports[`command init (app) it should work without init commit: fs 5`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/type",
}
`;

exports[`command init (app) it should work without init commit: fs 6`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/unit",
}
`;

exports[`command init (app) it should work without init commit: fs 7`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/integration",
}
`;

exports[`command init (app) it should work without init commit: fs 8`] = `
Object {
  "options": Object {
    "recursive": true,
  },
  "path": "/path/to/project/tests/e2e",
}
`;

exports[`command init (app) it should work without init commit: fs 9`] = `
Object {
  "data": "
",
  "path": "/path/to/project/src/lib/types.ts",
}
`;

exports[`command init (app) it should work without init commit: fs 10`] = `
Object {
  "data": "
",
  "path": "/path/to/project/tests/shared/.keep",
}
`;

exports[`command init (app) it should work without init commit: fs 11`] = `
Object {
  "data": "
",
  "path": "/path/to/project/tests/type/.keep",
}
`;

exports[`command init (app) it should work without init commit: fs 12`] = `
Object {
  "data": "{\\"file\\":\\"settings.json\\"}
",
  "path": "/path/to/project/.vscode/settings.json",
}
`;

exports[`command init (app) it should work without init commit: fs 13`] = `
Object {
  "data": "{\\"file\\":\\"carnarc.json\\"}
",
  "path": "/path/to/project/.carnarc.json",
}
`;

exports[`command init (app) it should work without init commit: fs 14`] = `
Object {
  "data": "editorconfig
",
  "path": "/path/to/project/.editorconfig",
}
`;

exports[`command init (app) it should work without init commit: fs 15`] = `
Object {
  "data": "eslintignore
",
  "path": "/path/to/project/.eslintignore",
}
`;

exports[`command init (app) it should work without init commit: fs 16`] = `
Object {
  "data": "gitignore
",
  "path": "/path/to/project/.gitignore",
}
`;

exports[`command init (app) it should work without init commit: fs 17`] = `
Object {
  "data": "npmignore
",
  "path": "/path/to/project/.npmignore",
}
`;

exports[`command init (app) it should work without init commit: fs 18`] = `
Object {
  "data": "prettierignore
",
  "path": "/path/to/project/.prettierignore",
}
`;

exports[`command init (app) it should work without init commit: fs 19`] = `
Object {
  "data": "// babel.config.js
",
  "path": "/path/to/project/babel.config.js",
}
`;

exports[`command init (app) it should work without init commit: fs 20`] = `
Object {
  "data": "{\\"file\\":\\"typescriptrc.json\\"}
",
  "path": "/path/to/project/tsconfig.json",
}
`;

exports[`command init (app) it should work without init commit: fs 21`] = `
Object {
  "data": "{\\"file\\":\\"typescriptrc.build.json\\"}
",
  "path": "/path/to/project/tsconfig.build.json",
}
`;

exports[`command init (app) it should work without init commit: fs 22`] = `
Object {
  "data": "// jest.config.js
",
  "path": "/path/to/project/jest.config.js",
}
`;

exports[`command init (app) it should work without init commit: fs 23`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/unit/dummy.test.ts",
}
`;

exports[`command init (app) it should work without init commit: fs 24`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/integration/dummy.test.ts",
}
`;

exports[`command init (app) it should work without init commit: fs 25`] = `
Object {
  "data": "// dummy.test.ts
",
  "path": "/path/to/project/tests/e2e/dummy.test.ts",
}
`;

exports[`command init (app) it should work without init commit: fs 26`] = `
Object {
  "data": "// dummyHook.test.ts
",
  "path": "/path/to/project/tests/e2e/pre.ts",
}
`;

exports[`command init (app) it should work without init commit: fs 27`] = `
Object {
  "data": "// dummyHook.test.ts
",
  "path": "/path/to/project/tests/e2e/post.ts",
}
`;

exports[`command init (app) it should work without init commit: fs 28`] = `
Object {
  "data": "{
  \\"name\\": \\"project-name\\",
  \\"license\\": \\"MIT\\",
  \\"scripts\\": {
    \\"prepare\\": \\"[ \\\\\\"$CI\\\\\\" != \\\\\\"\\\\\\" ] || husky install ./node_modules/carna/configs/husky\\",
    \\"analyse\\": \\"npx carna analyse\\",
    \\"build\\": \\"npx carna build\\",
    \\"prebuild\\": \\"rm -rf ./build\\",
    \\"test\\": \\"npx carna test\\",
    \\"test:watch\\": \\"npx carna test -w\\"
  },
  \\"dependencies\\": {
    \\"package\\": \\"0.1.0\\"
  },
  \\"main\\": \\"build/index.js\\",
  \\"types\\": \\"build/index.d.ts\\"
}
",
  "path": "/path/to/project/package.json",
}
`;
